transport/src/main/java/io/netty/channel/AbstractChannel.java

```diff
@@ -1048,4 +1065,56 @@ boolean setClosed() {
             return super.trySuccess();
         }
     }
+
+    private static final class AutoFlushTask implements Runnable {
+
+        private final AbstractChannel channel;
+        private final DefaultChannelPipeline pipeline;
+        private final SingleThreadEventLoop eventLoop;
+        private boolean writePending;
+        private boolean running;
+
+        AutoFlushTask(AbstractChannel channel, DefaultChannelPipeline pipeline) {
```

Comment:
I think you actually can access the `DefaultChannelPipeline` via channel.pipeline. So no need to pass it in here.